ludwig/visualize.py

```diff
@@ -221,18 +296,43 @@ def compare_classifiers_performance_from_prob_cli(
     :param ground_truth_split: (str) type of ground truth split -
         `0` for training split, `1` for validation split or
         2 for `'test'` split.
+    :param split_file: (str, None) file path to csv file containing split values
+    :param ground_truth_metadata: (str) file path to feature metadata json file
+        created during training.
     :param output_feature_name: (str) name of the output feature to visualize.
+    :param output_directory: (str) name of output directory containing training
+        results.
     :param kwargs: (dict) parameters for the requested visualizations.
 
     # Return
     :return None:
     """
-    gt = load_from_file(ground_truth, output_feature_name, ground_truth_split)
+
+    # retrieve feature metadata to convert raw predictions to encoded value
+    metadata = load_json(ground_truth_metadata)
+
+    # translate string to encoded numeric value
+    # retrieve ground truth from source data set
+    ground_truth = _extract_ground_truth_values(
+        ground_truth,
+        output_feature_name,
+        ground_truth_split,
+        split_file
+    )
+
+    feature_metadata = metadata[output_feature_name]
+    vfunc = np.vectorize(_encode_categorical_feature)
+    ground_truth = vfunc(ground_truth, feature_metadata['str2idx'])
```

Comment:
These 2 lines should likely happen inside the viz function,, not outside. The viz functin should accept a pandas column ground truth. If this means that you also need to pass the metadata for it work, we can add the metadata parameter. in that case i would rather have both fature name and metadata as parameters.